core/src/main/scala/pureconfig/CoproductHint.scala

```diff
@@ -92,6 +92,37 @@ class FieldCoproductHint[T](key: String) extends CoproductHint[T] {
 }
 
 /**
+ * Hint applicable to sealed families of case objects where objects are written and read as strings. Trying to read or
+ * write values that are not case objects results in failure.
+ *
+ * @tparam T the type of the coproduct or sealed family for which this hint applies
+ */
+class EnumCoproductHint[T] extends CoproductHint[T] {
+
+  /**
+   * Returns the field value for a class or coproduct option name.
+   *
+   * @param name the name of the class or coproduct option
+   * @return the field value associated with the given class or coproduct option name.
+   */
+  protected def fieldValue(name: String): String = name.toLowerCase
```

Comment:
The reason why this is protected is to give devs a hint at the fact that they can subclass this class and change how the `fieldValue` is calculated from a `name`?